docker/Dockerfile

```diff
@@ -0,0 +1,74 @@
+# Simplified mGBA Docker Environment for Testing
```

Comment:
this file is useless, lets remove it and rename and use dockerfile.complex mainly. rename dockerfile.complex to just dockerfile and make sure it is used with that name everywhere